src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -64,16 +69,36 @@ export const mapDefaultTokenToToggleable = <T extends Token>({
 }: {
 	defaultToken: T;
 	userToken: TokenToggleable<T> | undefined;
-}): TokenToggleable<T> => ({
-	...defaultToken,
-	enabled:
+}): TokenToggleable<T> => {
+	const isEnabledByDefault =
+		'ledgerCanisterId' in defaultToken &&
+		ICRC_CHAIN_FUSION_DEFAULT_LEDGER_CANISTER_IDS.includes(
+			(defaultToken as { ledgerCanisterId: CanisterIdText }).ledgerCanisterId
+		);
+
+	const isSuggestedToken =
 		('ledgerCanisterId' in defaultToken &&
-			ICRC_CHAIN_FUSION_DEFAULT_LEDGER_CANISTER_IDS.includes(
+			ICRC_CHAIN_FUSION_SUGGESTED_LEDGER_CANISTER_IDS.includes(
 				(defaultToken as { ledgerCanisterId: CanisterIdText }).ledgerCanisterId
 			)) ||
-		userToken?.enabled === true,
-	version: userToken?.version
-});
+		('address' in defaultToken &&
+			ERC20_SUGGESTED_TOKENS.map((token) => token.address).includes(
+				(
+					defaultToken as {
+						address: ContractAddressText;
+					}
+				).address
+			));
+
+	return {
+		...defaultToken,
+		enabled:
+			isEnabledByDefault ||
+			(isNullish(userToken?.enabled) && isSuggestedToken) ||
+			userToken?.enabled === true,
```

Comment:
`isNullish` is needed here?

Reply:
on line 97 It's needed - We only want to enable suggestedToken if the user has no preference at all.
on line 98 we're fine without isNullish, because: 
null === true => false
undefined === true => false
So no preference => no enabling

Reply:
thank you for the explanation!